### PR TITLE
Remove SSL Simulation Protocol and Fix Replay on master and Remove Ubuntu 18 support

### PR DESCRIPTION
**Pull request by itsarune**
### Description

omnibus bill that:
- fixes replay control on master when replay player wasn't shown
- removes SSL Simulation Protocol as a github submodule (https://github.com/UBC-Thunderbots/SSL-Simulation-Protocol-Support) -> I don't know why this exists and nothing uses it; it also doesn't compile properly
- remove ubuntu 18 support

### Testing Done

ran `bazel test //software...` locally

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[submodule "src/software/simulation/ssl_simulation_protocol"]
-	path = src/software/simulation/ssl_simulation_protocol
-	url = git@github.com:UBC-Thunderbots/SSL-Simulation-Protocol-Support.git
```
Comment (nimazareian) [inline .gitmodules:3]: Was this only for virtual robocup?

**Review by nimazareian — COMMENTED** [review]
Could you also update cpp test visualization? Seems to be facing the same problem.
https://github.com/UBC-Thunderbots/Software/blob/b29e0df7211e8557d8bfb209e72ebf0b2b15c8e7/src/software/thunderscope/thunderscope_main.py#L183-L186

I also noticed that the robot view widget is showing up for replays. The current logic just checks if both `load_blue` and `load_yellow` are set to True (assumes its AI vs AI) to not show it. 
https://github.com/UBC-Thunderbots/Software/blob/61b50f397b911a76b01a3c206a48c5e96c2eea19/src/software/thunderscope/thunderscope.py#L354-L356
We can fix it by adding another parameter to the Thunderscope constructor. But since we now have so many parameters and it's error prone, it might be a good idea (probably in a separate PR/ticket) to add a new Enum that we can pass to Thunderscope to select for which setting it should be configured. Then we can remove `load_blue`, `load_yellow`, ... and pass the initialization down to the Thunderscope. Though Im unsure if it will end up making it any nicer/more intuitive, given that there are so many different way that we can launch Thunderscope.

File: .gitmodules (hunk 1 above)
Comment (itsarune) [reply .gitmodules:3]: I can't explain to you why this exists. It seems to be used in our old simulator as a linking library between our firmware and our simulator (but then we changed simulators and firmware)

I went through our code history and I can't find a single commit that uses this git submodule

Comment (itsarune) [reply .gitmodules:3]: I removed it because it doesn't compile properly and prevents us from running `bazel test //software/...`

**Comment by nimazareian** [comment]
This bug seems to also be affecting pytest as well. This should be updated:
https://github.com/UBC-Thunderbots/Software/blob/ef0384d09186c55fe4f45d9a696a31538ee263ed/src/software/simulated_tests/simulated_test_fixture.py#L395-L401
to also have 
```py
load_blue=True,
load_yellow=True,
```

**Review by nimazareian — APPROVED** [review]
LGTM

**Review by williamckha — APPROVED** [review]

**Pull request merged**